File: graduation-study3/data/ikeda/dist.jl

# 制作途中
using Plots
using Printf
using TimeseriesSurrogates, CairoMakie
using PyCall

np = pyimport("numpy")
glob = pyimport("glob")
ordpy = pyimport("ordpy")
stats = pyimport("stats_func.py")
paths = glob.glob("original/u001/**.npy")
N = 100


for i in 1:N
    data = np.load(paths[1])
    x = data[:, 1]
    pats, probs = ordpy.ordinal_distribution(x, dx=6, return_missing=True)
    probs = sort_probs(pats, probs)
    np.save(f"dist/original/dist_{i}", probs)
end

File: graduation-study3/data/ikeda/dist.py




import glob 
import numpy as np
from ordpy import ordinal_distribution, ordinal_sequence
from stats_func import *
# import os



paths = glob.glob("aaft/u090/**.npy")
print(paths)

for i, path in enumerate(paths):
    data = np.load(path)
    x = data[:, 0]
    pats, probs = ordinal_distribution(x, dx=6, return_missing=True)
    probs = probs*len(ordinal_sequence(x, dx=6)) #回数を出す
    
    probs[probs<100] = 0
    probs = sort_probs(pats, probs)
    np.save(f"dist2/aaft/dist_{i}", probs)